lib/jekyll/document.rb

```diff
@@ -7,9 +7,9 @@ class Document
     attr_reader :path, :site, :extname, :collection
     attr_accessor :content, :output
 
-    YAML_FRONT_MATTER_REGEXP = /\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)/m
-    DATELESS_FILENAME_MATCHER = /^(.+\/)*(.*)(\.[^.]+)$/
-    DATE_FILENAME_MATCHER = /^(.+\/)*(\d+-\d+-\d+)-(.*)(\.[^.]+)$/
+    YAML_FRONT_MATTER_REGEXP = %r!\A(---\s*\n.*?\n?)^((---|\.\.\.)\s*$\n?)!m
+    DATELESS_FILENAME_MATCHER = %r!^(?:.+\/)*(.*)(\.[^.]+)$!
+    DATE_FILENAME_MATCHER = %r!^(?:.+\/)*(\d+-\d+-\d+)-(.*)(\.[^.]+)$!
```

Comment:
I added it to make the first group not captured, so that I can use all the captures later like this:
```ruby
if relative_path =~ DATE_FILENAME_MATCHER
  date, slug, ext = Regexp.last_match.captures
```
instead of
```ruby
if relative_path =~ DATE_FILENAME_MATCHER
   date, slug, ext = $2, $3, $4
```
So I changed the regexp to only capture date, slug and ext so that I don't need to use $2, $3, $4 (Rubocop doesn't like Pearl-style captures)